master/buildbot/test/unit/test_www_authz.py

```diff
@@ -140,3 +147,72 @@ def test_rebuildBuild(self):
         # not owner cannot rebuild
         yield self.assertUserForbidden("builds/13", "rebuild", {}, "eightuser")
         yield self.assertUserForbidden("buildrequests/82", "rebuild", {}, "eightuser")
+
+    @defer.inlineCallbacks
+    def test_fnmatchPatternRoleCheck(self):
+        # set defaultDeny to True so action is denied if no match
+        allow_rules = [
+                        AnyEndpointMatcher(role="[a,b]dmin?")
+        ]
+
+        self.setAllowRules(allow_rules)
+
+        yield self.assertUserAllowed("builds/13", "rebuild", {}, "homer")
+
+        # check if action is denied
+        with self.assertRaises(authz.Forbidden):
```

Comment:
this is not supported with tw14, please use assertRaisesRegex (which is monkey patched in buildbot)